src/orion/algo/tpe.py

```diff
@@ -518,18 +586,19 @@ def _sample_categorical_point(self, dimension, below_points, above_points):
 
         return new_point
 
-    def split_trials(self):
+    def split_trials(self) -> tuple[list[Trial], list[Trial]]:
         """Split the observed trials into good and bad ones based on the ratio `gamma``"""
 
-        trials = []
-        for trial, _ in self._trials_info.values():
+        trials: list[Trial] = []
+        for trial in self.registry:
             if trial.status != "completed":
                 trial = self.strategy.infer(trial)
 
             if trial is not None:
                 trials.append(trial)
-
-        sorted_trials = sorted(trials, key=lambda trial: trial.objective.value)
+        # NOTE: This assumes that all trials have an objective. Making assumption explicit.
+        assert all(trial.objective is not None for trial in trials)
```

Comment:
Like for Hyperband, I would filter in the loop above instead and log a warning if that happens.

Reply:
Like for HyperBand, here I'm only pointing out an implicit assumption of the existing code (sorting would raise an exception if any objective is `None`)
Improvements to the algorithms belong in another PR IMO.

Reply:
I agree it's outside the scope of this PR, but I have the same opinion as in the other discussion regarding the error message.

Reply:
I disagree. The code is explicit enough already, re-writing it in english next to it isn't good.

Reply:
Can we mark this as resolved please? Not going to budge on this. The TypeError is much less explicit than the assertion code itself.

Reply:
Deferring discussion here: https://github.com/Epistimio/orion/issues/859